### DISCUSSION TITLE
It is way too hard to delete a repository/please add a bulk delete

### DISCUSSION THREAD

**Discussion by josh-collinsworth** [community/community#75564] [Repositories]
### Select Topic Area

Product Feedback

### Body

I work on a SaaS product that creates repositories on behalf of users, corresponding to projects in the product. That means I pretty regularly have a lot of repos that need to be deleted.

The current UI workflow is:

1. Click into the repo
2. Click the "Settings" tab
3. Scroll all the way to the bottom of the page
4. Click "Delete this repository"
5. Click "I want to delete this repository"
6. Click "I have read and understand these effects"
7. Type what it tells you to into the box
8. Click "Delete this repository"

That's _way_ too many steps just to delete a private repo that doesn't even have any commits, especially since _four_ of them are pretty much exactly the same thing (clicking a red button that confirms your intent to delete the repository).

***Especially*** onerous: that the text in step 7 has apparently been made deliberately un-selectable, meaning you must manually type the text rather than copy and paste it. (Or resort to hacky workarounds like copying from dev tools, which usually takes just as long as typing anyway.)

It's so difficult I seriously want to build a Chrome extension just to hack through all the red tape.

I understand wanting to make it extremely difficult to delete a repository by mistake, and to make it extremely clear what will happen when the process is done. But the number of steps to delete a repo has ballooned to the excessive. Preventing users from copying the repo name to paste into the confirmation input, especially, does not seem so much productive as exceptionally frustrating.

Reasonable friction would be understandable. As it is now, however, the process is stuffed with unreasonable, artificially manufactured frustration.

Suggestions:

- Remove at least one of the steps above
- Make the steps dependent on traffic or recent activity
- Allow users to (temporarily?) enable an easier way to bulk delete repos in situations like mine

**Comment by domenic** [comment]
If you use the GitHub CLI it's pretty easy: `gh repo delete --yes domenic/my_crappy_fork`

**Reply by taylorjayoung** [reply]
I built reposweeper.com for this. It's free to use and lets you bulk delete in an intuitive UI.  @josh-collinsworth @domenic 

Sorry for the delay!

**Reply by taylorjayoung** [reply]
This is implemented on reposweeper.com now. Just basic select branches in bulk and click delete. But lmk if you have any feedback or suggestions for other bulk branch management features. Cheers @domenic @josh-collinsworth 

<img width="1236" height="725" alt="ManageBranches" src="https://github.com/user-attachments/assets/1b4ea005-3c19-4731-a3f8-e1467aeb40c4" />
<img width="921" height="782" alt="Delete Instructions" src="https://github.com/user-attachments/assets/e1078137-4837-46ec-9e6f-672397ca6c28" />

**Reply by domenic** [reply]
Please stop @-mentioning me.

**Comment by SIMARSINGHRAYAT** [comment]
Great feedback on repository deletion complexity! Here are the practical solutions mentioned:

**Immediate Solutions Available:**

1. **GitHub CLI** (Fastest for bulk deletion)
   ```
   gh repo delete --yes org/repo-name
   ```
   Perfect for your SaaS use case - automate it in your platform!

2. **Third-party Tools**
   - reposweeper.com - Free, intuitive UI for bulk deletion
   - Could integrate this into your platform

3. **GitHub API**
   - Programmatic deletion via REST API
   - Best for enterprise automation

**Why Current UI is Restrictive:**
GitHub intentionally adds friction to prevent accidental deletion. While this is good for individual users, it's painful for platforms creating/managing many repos.

**Recommendation for your SaaS:**
Instead of the web UI, integrate GitHub CLI or the API into your platform's admin panel. This gives you one-click deletion while maintaining safety through your own confirmation.

**For GitHub to Consider:**
Allow users who create repos programmatically (via API/CLI) to also delete them via the same method without the Web UI friction.

**Comment by Chintanpatel24** [comment]
i think by using the GITHUB CLI or TOKEN you can do this
